### PR TITLE
Fix v6 thunk nixpkgs fetching

### PR DESCRIPTION
**Pull request by plt-amy**
v6 thunks do not import their nixpkgs so they can not be used outside of the very specific case of fetching GitHub repos that have no submodules.

### CODE REVIEW

**Review by madeline-os — APPROVED** [review]
Oof. I hope no one's upgraded yet!

File: default.nix [hunk 1]
```diff
@@ -64,7 +67,19 @@ in rec {
         ver = "0.9.1";
         sha256 = "152lnv339fg8nacvyhxjfy2ylppc33ckb6qrgy0vzanisi8pgcvd";
       } {};
-      nix-thunk = self.callCabal2nix "nix-thunk" (gitignoreSource ./.) {};
+      nix-thunk = pkgs.haskell.lib.overrideCabal (self.callCabal2nixWithOptions "nix-thunk" (gitignoreSource ./.) "-f is-nix-build" {}) {
+        librarySystemDepends = with pkgs; [
+          (writeTextFile {
+            name = "print-nixpkgs-path";
```
Comment (madeline-os) [inline default.nix:81]: swag

Comment (madeline-os) [reply default.nix:81]: comment, though

File: dep/which/thunk.nix [hunk 2]
```diff
@@ -2,7 +2,7 @@
 let fetch = { private ? false, fetchSubmodules ? false, owner, repo, rev, sha256, ... }:
   if !fetchSubmodules && !private then builtins.fetchTarball {
     url = "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz"; inherit sha256;
-  } else (import <nixpkgs> {}).fetchFromGitHub {
+  } else (import "/nix/store/qjg458n31xk1l6lj26c3b871d4i4is98-source" {}).fetchFromGitHub {
```
Comment (madeline-os) [inline dep/which/thunk.nix:5]: Is this what the pinned nixpkgs evaluates to? Seems so.

**Review by madeline-os — CHANGES_REQUESTED** [review]
Just want the dependency trick documented in a comment

**Review by madeline-os — APPROVED** [review]
Addressed my feedback before I even gave it =.=

File: dep/which/thunk.nix (hunk 2 above)
Comment (plt-amy) [reply dep/which/thunk.nix:5]: Yeah I did
```fish
for x in dep/*/thunk.nix; nix-thunk unpack (dirname $x); and result/bin/nix-thunk pack (dirname $x); end
```
to update the thunks

**Pull request merged**